core/modules/wildcard_match.cc

```diff
@@ -335,7 +335,10 @@ int WildcardMatch::DelEntry(struct WmTuple *tuple, wm_hkey_t *key) {
     tuple->ht.Close();
 
     num_tuples_--;
-    memmove(&tuples_[idx], &tuples_[idx + 1],
+
+    // shift remaing tuples to the left
+    memmove(reinterpret_cast<void*>(&tuples_[idx]),
```

Comment:
Kind of tangential, but it would be nice if we had a way to measure the impact of these sorts of changes. Individual module benchmarks seems kind of tedious, so maybe just for those that ought to be fast-- or this might also just be something we punt to integration testing.